File: src/main/scala/qmf/poc/agent/catalog/models/ObjectData.scala
package qmf.poc.agent.catalog.models

import spray.json.{JsNumber, JsObject, JsString, JsValue, JsonFormat}

class ObjectData(val owner: String, val name: String, val `type`: String, val seq: Short, val appldata: String)

object ObjectData:
  given JsonFormat[ObjectData] with
    def write(o: ObjectData): JsObject =
      JsObject(
        "name" -> JsString(o.name),
        "owner" -> JsString(o.owner),
        "seq" -> JsNumber(o.seq),
        "type" -> JsString(o.`type`),
        "appldata" -> JsString(o.appldata)
      )

    //noinspection NotImplementedCode
    def read(value: JsValue): ObjectData = ???

File: src/main/java/qmf/poc/agent/catalog/models/ObjectData.java
package qmf.poc.agent.catalog.models;

public class ObjectData {
    public final String owner;
    public final String name;
    public final String type;
    public final String appldata;

    public ObjectData(String owner, String name, String type, String appldata) {
        this.owner = owner;
        this.name = name;
        this.type = type;
        this.appldata = appldata;
    }

    @Override
    public String toString() {
        final String s = "'" + owner + "'|'" + name + "'|'" + type + "'|'" + appldata;// + "'";

        if (s.length() < 131) {
            return s + "'";
        } else {
            return s.substring(0, 128) + "...'";
        }
    }
}
